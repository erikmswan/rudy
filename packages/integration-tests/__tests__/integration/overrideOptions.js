import {
  compose,
  createHistory,
  createRequest,
  createReducer,
} from '@respond-framework/rudy/src/core'
import shouldTransition from '@respond-framework/rudy/src/utils/shouldTransition'
import shouldCall from '@respond-framework/rudy/src/middleware/call/utils/shouldCall'
import createTest from '../../__helpers__/createTest'

createTest(
  'core capabilities can be overriden as options',
  {
    SECOND: {
      path: '/second',
      thunk() {},
    },
  },
  {
    createHistory: (...args) => createHistory(...args),
    createReducer: (...args) => createReducer(...args),
    compose: (...args) => compose(...args),
    shouldTransition: (...args) => shouldTransition(...args),
    createRequest: (...args) => createRequest(...args),
    shouldCall: (...args) => shouldCall(...args),
    location: 'location',
  },
)
