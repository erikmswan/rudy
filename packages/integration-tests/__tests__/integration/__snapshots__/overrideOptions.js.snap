// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`core capabilities can be overriden as options: 1 - options - compose 1`] = `2`;

exports[`core capabilities can be overriden as options: 1 - options - createHistory 1`] = `1`;

exports[`core capabilities can be overriden as options: 1 - options - createReducer 1`] = `1`;

exports[`core capabilities can be overriden as options: 1 - options - createRequest 1`] = `2`;

exports[`core capabilities can be overriden as options: 1 - options - shouldCall 1`] = `12`;

exports[`core capabilities can be overriden as options: 1 - response 1`] = `
Object {
  "basename": "",
  "hash": "",
  "location": Object {
    "blocked": null,
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/first",
          "scene": "",
          "search": "",
          "url": "/first",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "FIRST",
      },
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/second",
          "scene": "",
          "search": "",
          "url": "/second",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "SECOND",
      },
    ],
    "from": null,
    "index": 1,
    "key": "345678",
    "kind": "push",
    "length": 2,
    "n": 1,
    "pathname": "/second",
    "pop": false,
    "prev": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": 0,
        "key": "345678",
        "pathname": "/first",
        "scene": "",
        "search": "",
        "url": "/first",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "FIRST",
    },
    "scene": "",
    "search": "",
    "status": 200,
    "url": "/second",
  },
  "params": Object {},
  "query": Object {},
  "state": Object {},
  "type": "SECOND",
}
`;

exports[`core capabilities can be overriden as options: 1 - routes - NEVER_USED_PATHLESS - thunk 1`] = `0`;

exports[`core capabilities can be overriden as options: 1 - routes - REDIRECTED - onComplete 1`] = `0`;

exports[`core capabilities can be overriden as options: 1 - routes - SECOND - thunk 1`] = `1`;

exports[`core capabilities can be overriden as options: 1 - state 1`] = `
Object {
  "location": Object {
    "basename": "",
    "blocked": null,
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/first",
          "scene": "",
          "search": "",
          "url": "/first",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "FIRST",
      },
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/second",
          "scene": "",
          "search": "",
          "url": "/second",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "SECOND",
      },
    ],
    "from": null,
    "hash": "",
    "index": 1,
    "key": "345678",
    "kind": "push",
    "length": 2,
    "n": 1,
    "params": Object {},
    "pathname": "/second",
    "pop": false,
    "prev": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": 0,
        "key": "345678",
        "pathname": "/first",
        "scene": "",
        "search": "",
        "url": "/first",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "FIRST",
    },
    "query": Object {},
    "scene": "",
    "search": "",
    "state": Object {},
    "status": 200,
    "type": "SECOND",
    "universal": false,
    "url": "/second",
  },
  "title": "SECOND",
}
`;

exports[`core capabilities can be overriden as options: 1 - title 1`] = `"SECOND"`;
